### ISSUE TITLE
GET v1/stores 500 (Internal Server Error)

### ISSUE THREAD

**Issue by cwackerfuss** [blockstack/blockstack-core#690]
In addition to the bug we see in https://github.com/blockstack/blockstack-core/issues/689, we've been getting several complaints about GET requests resulting in a 500 server error. This seems to be happening on several apps being developed. Here are a few complaints I found through Slack:

<img width="379" alt="screen shot 2017-11-28 at 12 44 06 pm" src="https://user-images.githubusercontent.com/4405356/33338050-76f9b7ee-d43a-11e7-895b-95bdb334e495.png">

<img width="599" alt="screen shot 2017-11-28 at 12 43 34 pm" src="https://user-images.githubusercontent.com/4405356/33338077-8a7bd4c8-d43a-11e7-94c5-39588502bfde.png">

<img width="549" alt="screen shot 2017-11-28 at 12 43 16 pm" src="https://user-images.githubusercontent.com/4405356/33338093-90a89354-d43a-11e7-8a8a-067a551e0378.png">

<img width="1238" alt="screen shot 2017-11-28 at 12 49 49 pm" src="https://user-images.githubusercontent.com/4405356/33338117-a7d2bcbc-d43a-11e7-8155-3a7c17d88581.png">

**Comment by cwackerfuss** [comment]
@larrysalibra @jcnelson @kantai

**Comment by dantrevino** [comment]
I was able to create a 500 internal server error by doing rapid-fire putFile() without waiting for the promise return.  I was able to recreate this using a blockstack id with and without an associated username.

my code:
```
      blockstack.putFile(STORAGE_FILE, JSON.stringify(this.transactions))
      console.log('writing file: ' + JSON.stringify(this.transactions))

```
console output: https://gist.github.com/dantrevino/334d36b72cef1cc4bc014e6b1039ffca

There are more than  4 successful writes before the error appeared, but I truncated the log.
